### PR TITLE
Initial implementation of e2e tests

### PR DESCRIPTION
**Pull request by JustinKuli**
Just testing basic functionality. Many helper functions are copied from
similar repositories, although I simplified some things for this set.

Refs:
 - https://github.com/stolostron/backlog/issues/24299

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

### CODE REVIEW

File: test/e2e/basic_test.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+const (
+	adminIAMPolicyYaml string = "../resources/iam-policy-default.yaml"
+	adminIAMPolicyName string = "rule-of-two"
+	adminCRBYaml       string = "../resources/admin-crb.yaml"
+	otherIAMPolicyYaml string = "../resources/iam-policy-other.yaml"
+	otherIAMPolicyName string = "there-can-be-only-one"
+	otherCRBYaml       string = "../resources/other-crb.yaml"
+	otherRoleYaml      string = "../resources/other-clusterrole.yaml"
+)
```
Comment (dhaiducek) [inline test/e2e/basic_test.go:20]: Would you please make these block scoped variables? I'm not sure there's a need to make them package-level.

File: test/resources/other-crb.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: insert-story-reference
+roleRef:
+  name: immortal
```
Comment (dhaiducek) [inline test/resources/other-crb.yaml:6]: Haha! Nice.

File: test/e2e/basic_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+var _ = Describe("Test IAM policy", Ordered, func() {
+	Describe("Test default clusterrole (cluster-admin)", func() {
```
Comment (dhaiducek) [inline test/e2e/basic_test.go:23]: Nit: Should these tests be in a loop since at quick glance I think they're doing the same/similar things?

**Review by dhaiducek — CHANGES_REQUESTED** [review]
Thanks for getting these E2E tests going! Looks great--I have some minor comments.

File: test/e2e/basic_test.go (hunk 3 above)
Comment (JustinKuli) [reply test/e2e/basic_test.go]: It didn't feel quite right in a loop/table, but I was able to pull some of the common stuff out. Let me know what you think of this approach (and thanks for the suggestion :).

**Review by dhaiducek — APPROVED** [review]
LGTM! Thanks for tackling this! 😄

**Pull request merged**